src/apps/sturdy/ethereum/sturdy.lending.token-fetcher.ts

```diff
@@ -0,0 +1,106 @@
+import { Inject } from '@nestjs/common';
+import axios from 'axios';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { Register } from '~app-toolkit/decorators';
+import {
+  buildNumberDisplayItem,
+  buildPercentageDisplayItem,
+} from '~app-toolkit/helpers/presentation/display-item.present';
+import { getImagesFromToken } from '~app-toolkit/helpers/presentation/image.present';
+import { ContractType } from '~position/contract.interface';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { BaseToken } from '~position/token.interface';
+import { Network } from '~types/network.interface';
+
+import { SturdyContractFactory } from '../contracts';
+import { STURDY_DEFINITION } from '../sturdy.definition';
+
+const appId = STURDY_DEFINITION.id;
+const groupId = STURDY_DEFINITION.groups.lending.id;
+const network = Network.ETHEREUM_MAINNET;
+
+type VaultMonitoringResponse = {
+  chain: string;
+  tokens: string;
+  decimals: number;
+  address: string;
+  supply: number;
+  price: number;
+  base: number;
+  reward: number;
+  rewards: {
+    CRV: number;
+    CVX: number;
+  };
+  url: number;
+  tvl: number;
+  active: boolean;
+}[];
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class EthereumSturdyLendingTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(SturdyContractFactory) private readonly sturdyContractFactory: SturdyContractFactory,
+  ) {}
+
+  async getPositions() {
+    const multicall = this.appToolkit.getMulticall(network);
+    const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
+    const ethToken = baseTokens.find(t => t.address === ZERO_ADDRESS);
+    if (!ethToken) return [];
+
+    const endpoint = 'https://us-central1-stu-dashboard-a0ba2.cloudfunctions.net/getVaultMonitoring?chain=ethereum';
+    const tokenData = await axios.get<VaultMonitoringResponse>(endpoint).then(v => v.data);
```

Comment:
You can extract these out into a private method annotated with the `@CacheOnInterval` decorator to reduce the number of calls we make to this API. This caching helps protocol developers keep their costs down so we don't hammer their APIs with requests!

You'll see an example in `src/apps/good-ghosting/helpers/good-ghosting.game.config-fetcher.ts`